okta/komand_okta/actions/update_blacklist_zones/action.py

```diff
@@ -0,0 +1,96 @@
+import komand
+from .schema import UpdateBlacklistZonesInput, UpdateBlacklistZonesOutput, Input, Output, Component
+# Custom imports below
+import validators
+import requests
+from komand.exceptions import PluginException
+
+
+class UpdateBlacklistZones(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name='update_blacklist_zones',
+            description=Component.DESCRIPTION,
+            input=UpdateBlacklistZonesInput(),
+            output=UpdateBlacklistZonesOutput())
+
+    def run(self, params={}):
+        name = params.get(Input.NAME)
+        value = params.get(Input.ADDRESS)
+
+        if validators.ipv4_cidr(value) or validators.ipv6_cidr(value):
+            value_type = "CIDR"
+        else:
+            value_type = "RANGE"
+
+        url = requests.compat.urljoin(
+            self.connection.okta_url,
+            '/api/v1/zones'
+        )
+
+        response = self.connection.session.get(url)
+        zones = response.json()
+        zone = None
+        for search_zone in zones:
+            if search_zone.get("name") == name:
+                zone = search_zone
+                break
+
+        if not zone:
+            raise PluginException(cause=f"Name {name} does not exist in Okta zones. ",
```

Comment:
You don't need a space at the end. SDK handles that

```suggestion
            raise PluginException(cause=f"Name {name} does not exist in Okta zones. ",
```